src/components/arpfingerprint/arpfingerprint.tsx

```diff
@@ -0,0 +1,156 @@
+    const [isScanning, setIsScanning] = useState(false); // State hook for scanning status
+    const [pidScan, setPidScan] = useState(""); // State hook for process ID of the scanning process
+    const [allowSave, setAllowSave] = useState(true); // State hook for allowing output saving
+    const [hasSaved, setHasSaved] = useState(false); // State hook for tracking if output has been saved
+    const [output, setOutput] = useState(""); // State hook for storing the output of the scanning process
+
+    // Section: Form Handling
+    let form = useForm({
+        initialValues: {
+            targetIP: "",
+        },
+    });
+
+    // Section: Callback Functions
+
+    /**
+     * Callback function to handle process data received during scanning
+     * @param {string} data - Data received from the scanning process
+     */
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data); // Update output
+    }, []);
+
+    /**
+     * Callback function to handle process termination
+     * @param {Object} - Object containing exit code and signal code
+     */
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            if (code === 0) {
+                handleProcessData("\nScan completed successfully.");
+            } else if (signal === 15) {
+                handleProcessData("\nScan was manually terminated.");
+            } else {
+                handleProcessData(`\nScan terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+            setPidScan(""); // Clear the process id reference
+            setIsScanning(false); // Disable scanning state
+            setAllowSave(true); // Allow saving output
+            setHasSaved(false); // Reset save status
+        },
+        [handleProcessData]
+    );
+
+    /**
+     * Callback function to handle completion of output saving
+     */
+    const handleSaveComplete = () => {
+        setHasSaved(true); // Set save status to true after saving
+        setAllowSave(true); // Disable saving option
+    };
+
+    /**
+     * Callback function to clear output
+     */
+    const clearOutput = useCallback(() => {
+        setOutput(""); // Clear output
+        setHasSaved(false); // Reset save status
+        setAllowSave(false); // Disable saving option
+    }, [setOutput]);
+
+    // Section: Form Submission
+
+    /**
+     * Function to handle form submission
+     * Initiates the ARP fingerprinting process based on the provided IP address
+     * @param {FormValues} values - Form input values containing the target IP address
+     */
+    const onSubmit = async (values: FormValues) => {
+        const args = [values.targetIP]; // Command arguments
+
+        try {
+            // Execute the command with elevated privileges using CommandHelper
+            const result = await CommandHelper.runCommandWithPkexec(
```

Comment:
Please confirm that you need elevated privileges to run this command. Otherwise, change to runCommandGetPidAndOutput.